React/Base/RCTDefines.h

```diff
@@ -99,3 +99,10 @@ _Pragma("clang diagnostic ignored \"-Wunused-parameter\"") \
 RCT_EXTERN NSException *_RCTNotImplementedException(SEL, Class); \
 method NS_UNAVAILABLE { @throw _RCTNotImplementedException(_cmd, [self class]); } \
 _Pragma("clang diagnostic pop")
+
+/**
+ * Add the default Metro packager port number 
+ */
+#ifndef RCT_METRO_PORT
+#define RCT_METRO_PORT 8081
+#endif
```

Comment:
Maybe add a new line at the end of the file? It does cause a clang warning, maybe that's the issue.

```
In file included from /Users/distiller/react-native/React/Base/RCTConvert.h:15:
/Users/distiller/react-native/React/Base/RCTDefines.h:108:7: warning: no newline at end of file [-Wnewline-eof]
#endif
      ^
/Users/distiller/react-native/React/Base/RCTBundleURLProvider.m:17:67: error: expected expression
const NSUInteger kRCTBundleURLProviderDefaultPort = RCT_METRO_PORT;
```

Reply:
Thats just a warning and seems to have no effect - the error is in the preprocessor replacement which isn't replaced.  This has to be something unique the CircleCI compiler...

Reply:
Have you tried running `./scripts/objc-test-ios.sh`? This is the script CI runs.

Reply:
@janicduplessis thank you!  It took a few days (literally) but I finally figured this out.   xcode was setting `RCT_METRO_PORT` to empty string on the testing, so I figured out how to test for that and then redefine to 8081 if its empty.  That was a sporty problem 👍